src/v/net/tls.h

```diff
@@ -12,9 +12,55 @@
 #include "base/seastarx.h"
 
 #include <seastar/core/future.hh>
-
+#include <seastar/net/tls.hh>
 namespace net {
 
+/**
+ * Either a path to certificate file or the certificate content itself
+ * in PEM format.
+ */
+using certificate = std::variant<std::filesystem::path, ss::sstring>;
+
+/**
+ * Path to key/cert files
+ */
+struct key_cert_path {
+    std::filesystem::path key;
+    std::filesystem::path cert;
+};
+/**
+ * Key/cert pair content in PEM format
+ */
+struct key_cert {
+    ss::sstring key;
+    ss::sstring cert;
+};
+
+/**
+ * PKCS#12 container with certificate and private key
+ */
+struct pkcs12 {
+    certificate cert;
+    ss::sstring password;
+};
+
+/**
+ * Keystore can be either a path or content of key/cert files or a PKCS#12
+ * container
+ */
+using key_store = std::variant<key_cert, key_cert_path, pkcs12>;
+
+/**
+ * Returns credentials builder based on the provided configuration
+ */
+ss::future<ss::tls::credentials_builder> get_credentials_builder(
+  std::optional<certificate> trust_store,
+  std::optional<key_store> k_store,
+  std::optional<certificate> crl,
+  ss::tls::tls_version tls_version,
+  bool enable_renegotiation,
+  bool require_client_auth);
```

Comment:
This is great, however it's very hard to read at callsites when looking in github, etc. Can we use a parameters struct to make this more readable?

For example, right now this is what a caller looks like:

```
get_credentials_builder(std::nullopt, std::nullopt, std::nullopt, config::shard_local_cfg().tls_min_version, true, false);
```

Where instead it could be:

```
get_credentials_builder({
  .trust_store = std::nullopt, // or even omit!
  .k_store = std::nullopt,
  .crl = std::nullopt,
  .tls_version = ...,
  .enable_renegotiation = true,
  .require_client_auth = false,
});
```

Reply:
agreed, it will be more readable will do